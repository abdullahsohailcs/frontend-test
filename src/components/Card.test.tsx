import React from 'react';
import { render } from '@testing-library/react';
import Card from './Card';

// Mocking the WebPointer component as we don't need to test it in this snapshot
jest.mock('./svg/webpointer', () => () => <div>WebPointer</div>);

describe('Card', () => {
  it('should match the snapshot', () => {
    const props = {
      title: 'Sample Card Title',
      description: 'This is a description for the sample card.',
      image: 'https://via.placeholder.com/150',
      url: 'https://www.example.com',
    };

    // Render the Card component
    const { asFragment } = render(<Card {...props} />);

    // Match the snapshot
    expect(asFragment()).toMatchSnapshot();
  });
});
