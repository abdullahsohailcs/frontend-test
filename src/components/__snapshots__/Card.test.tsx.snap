// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component renders correctly and matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="card"
  >
    <div
      class="card-image"
    >
      <img
        alt="Docker Swarm"
        src="https://via.placeholder.com/100"
      />
    </div>
    <div
      class="card-content"
    >
      <h2
        class="card-title"
      >
        Docker Swarm
      </h2>
      <p
        class="card-description"
      >
        Native clustering for Docker. Turn a pool of Docker hosts into a single, virtual host.
      </p>
    </div>
    <a
      class="card-link"
      href="https://github.com/docker-archive/classicswarm"
      rel="noopener noreferrer"
      target="_blank"
    >
      <svg
        fill="none"
        height="26"
        viewBox="0 0 26 26"
        width="26"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M24.5694 1.43021C23.6163 0.476795 22.4678 0 21.1251 0H4.87497C3.53221 0 2.3838 0.476795 1.43021 1.43021C0.476795 2.3838 0 3.53215 0 4.87497V21.1249C0 22.4676 0.476795 23.6161 1.43021 24.5697C2.3838 25.5233 3.53221 26 4.87497 26H21.1249C22.4677 26 23.6161 25.5233 24.5693 24.5697C25.5228 23.6161 25.9996 22.4677 25.9996 21.1249V4.87497C25.9996 3.53215 25.5228 2.38362 24.5694 1.43021ZM21.6666 13.5415C21.6666 14.0155 21.4463 14.3482 21.0064 14.5402C20.8599 14.5966 20.7183 14.6248 20.5831 14.6248C20.2787 14.6248 20.0248 14.5174 19.8215 14.3032L17.3841 11.8658L8.34546 20.9049C8.13078 21.1192 7.87709 21.2266 7.58361 21.2266C7.29001 21.2266 7.03608 21.1192 6.8217 20.9049L5.09517 19.1781C4.88072 18.9641 4.77347 18.7097 4.77347 18.4164C4.77347 18.1233 4.88072 17.8693 5.09517 17.655L14.1344 8.61569L11.6968 6.17824C11.3469 5.85097 11.268 5.45623 11.4598 4.99337C11.6515 4.55339 11.9846 4.33313 12.4585 4.33313H20.5832C20.8766 4.33313 21.1305 4.44044 21.3453 4.65495C21.5595 4.86922 21.6667 5.12321 21.6667 5.41662V13.5415H21.6666Z"
          fill="#999FAA"
          opacity="0.2"
        />
        <h1>
          hh
        </h1>
      </svg>
    </a>
  </div>
</DocumentFragment>
`;

exports[`Card should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="card"
  >
    <div
      class="card-image"
    >
      <img
        alt="Sample Card Title"
        src="https://via.placeholder.com/150"
      />
    </div>
    <div
      class="card-content"
    >
      <h2
        class="card-title"
      >
        Sample Card Title
      </h2>
      <p
        class="card-description"
      >
        This is a description for the sample card.
      </p>
    </div>
    <div
      class="card-webpointer-container"
    >
      <a
        class="card-link"
        href="https://www.example.com"
        rel="noopener noreferrer"
        target="_blank"
      >
        <div>
          WebPointer
        </div>
      </a>
    </div>
  </div>
</DocumentFragment>
`;
